src/com/lms/model/Loan.java

```diff
@@ -60,50 +49,21 @@ public void setDueDate(LocalDate dueDate) {
 
 	@Override
 	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((book == null) ? 0 : book.hashCode());
-		result = prime * result + ((borrower == null) ? 0 : borrower.hashCode());
-		result = prime * result + ((branch == null) ? 0 : branch.hashCode());
-		result = prime * result + ((dateOut == null) ? 0 : dateOut.hashCode());
-		result = prime * result + ((dueDate == null) ? 0 : dueDate.hashCode());
-		return result;
+		return Objects.hash(book, borrower, branch);
 	}
 
 	@Override
-	public boolean equals(Object obj) {
-		if (this == obj)
+	public boolean equals(final Object obj) {
+		if (this == obj) {
 			return true;
-		if (obj == null)
-			return false;
-		if (getClass() != obj.getClass())
-			return false;
-		Loan other = (Loan) obj;
-		if (book == null) {
-			if (other.book != null)
-				return false;
-		} else if (!book.equals(other.book))
-			return false;
-		if (borrower == null) {
-			if (other.borrower != null)
-				return false;
-		} else if (!borrower.equals(other.borrower))
-			return false;
-		if (branch == null) {
-			if (other.branch != null)
-				return false;
-		} else if (!branch.equals(other.branch))
-			return false;
-		if (dateOut == null) {
-			if (other.dateOut != null)
-				return false;
-		} else if (!dateOut.equals(other.dateOut))
-			return false;
-		if (dueDate == null) {
-			if (other.dueDate != null)
-				return false;
-		} else if (!dueDate.equals(other.dueDate))
+		} else if (obj instanceof Loan) {
+			return Objects.equals(book, ((Loan) obj).getBook())
+					&& Objects.equals(borrower, ((Loan) obj).getBorrower())
```

Comment:
How is the Objects.eqauals able to compare Borrower with Borrower?

Reply:
`Objects.equals` is a helper method provided by the SDK that basically boils down to the following algorithm:

- If either argument is null, return true if the other is also null and false if it is not null
- Otherwise, return the result of `first.equals(second)`.

Reply:
Okay, that makes sense. So, it would use the functions that we defined in the other models.